### DISCUSSION TITLE
Visual scripting with graphical-based diffing

### DISCUSSION THREAD

**Discussion by michaelhoehn** [BHoM/BHoM#1141] [Ideas]
Hello 👋 

As more people adopt visual scripting languages in their work, it will become increasingly important to track changes beyond file versions (standard Sharepoint version tracking for example). 

The latest Diffing_Engine work is really fantastic and useful for many things such as object property diffing. I'm wondering though if there's a possibility of combining the great work done within that engine and implementing it on entire grasshopper files. I imagine the result looking like an image or diagram, or even text-based diff. Could also utilise similar functionality to that of metahopper in hacking the UI to 'group' or colour the deltas on the canvas. 

This would be super helpful as an instruction tool, especially for beginners, or additionally in configurations of whole workflows. As changes to workflows are quite common at times as well. 

***
Some references:
Graphical diffing in grasshopper and dynamo - https://github.com/oderby/VVD
MetaHopper - https://www.food4rhino.com/app/metahopper (see video at timestamp 2:45-3:10)

**Comment by BingWangUS** [comment]
I think this will be helpful when we are considering using Github for version control of our example workflow scripts. Great idea @michaelhoehn
